codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -445,6 +434,20 @@ private static MqttFixedHeader decodeFixedHeader(ByteBuf buffer) {
         return new Result<String>(s, numberOfBytesConsumed);
     }
 
+    private static Result<byte[]> decodeByteArray(ByteBuf buffer) {
+        final Result<Integer> decodedSize = decodeMsbLsb(buffer);
+        int size = decodedSize.value;
+        int numberOfBytesConsumed = decodedSize.numberOfBytesConsumed;
+        if (size < 0) {
+            numberOfBytesConsumed += size;
```

Comment:
I still don't understand this line of code. 2 bytes have still been consumed either way even if the data is invalid. This value is ultimately set to `bytesRemainingInVariablePart` which may be used later during the decode process. You mentioned this was a [hypothetical situation](https://github.com/netty/netty/pull/6817#discussion_r120771580) but it is not clear what will happen if this situation does occur (if its possible at all). If this situation is possible please add a unit test, and if it isn't possible can we remove this line?